frontend/src/core/editor/selectors.js

```diff
@@ -62,19 +62,26 @@ export function _existingTranslation(
         }
         // If translation is a string, from the generic editor.
         else {
-            // Except it might actually be a simple Fluent message from the SimpleEditor.
+            // Except it might actually be a Fluent message from the Simple or Source
+            // editors.
             if (entity.format === 'ftl') {
-                // This case happens when the format is Fluent and the string is simple.
-                // We thus store a simplified version of the Fluent message in memory,
-                // but the history has actual Fluent-formatted strings (with ID).
-                // Thus, we need to reconstruct the Fluent message in order to be able
-                // to compare it with items in history.
-                const reconstructed = fluent.serializer.serializeEntry(
-                    fluent.getReconstructedMessage(entity.original, translation)
-                );
+                // For Fluent files, the translation can be stored as a simple string
+                // when the Source editor or the Simple editor are on. Because of that,
+                // we want to turn the string into a Fluent message, as that's simpler
+                // to handle and less prone to errors. We do the same for each history
+                // entry.
+                let ftlMessage = fluent.parser.parseEntry(translation);
```

Comment:
Nit: I don't like that we're using two inconsistent names for the seamingly same concept: `ftlMessage` and `fluentTranslation`.